lib/rucio/web/ui/main.py

```diff
@@ -352,6 +361,62 @@ def GET(self):  # pylint:disable=no-self-use,invalid-name
         return check_token(render.subscriptions_editor())
 
 
+class LoadLogfile():
+    """ Loads logfile content list """
+    def GET(self):
+        try:
+            data = web.input()
+            response = requests.get(str(data.file_location), cert=config_get('webui', 'usercert'), verify=False)   # TODO: cert to ddmadmin
+            cont = response.content
+            file_like_object = io.BytesIO(cont)
+            tar = tarfile.open(mode='r:gz', fileobj=file_like_object)
+            jsonResponse = {}
+            for member in tar.getmembers():
+                jsonResponse[member.name] = member.size
+            web.header('Content-Type', 'application/json')
+            return json.dumps(jsonResponse)
+        except Exception, e:
+            print e
+            e_type = sys.exc_info()[0]
+            e_value = sys.exc_info()[1]
+            e_traceback = sys.exc_info()[2]
+            return 'Error: ' + str(e_type) + ' ' + str(e_value) + ' ' + str(e_traceback)
+
+
+class ExtractLogfile():
+    """ Extracts selected logfile content """
+    def GET(self):
+        try:
+            pyDict = {}
+            data = web.input()
+            response = requests.get(str(data.file_location), cert=config_get('webui', 'usercert'), verify=False)   # TODO: cert to ddmadmin
+            cont = response.content
+            file_like_object = io.BytesIO(cont)
+            tar = tarfile.open(mode='r:gz', fileobj=file_like_object)
+            for member in tar.getmembers():
+                if member.name == str(data.file_name):
+                    try:
+                        f = tar.extractfile(member)
+                        pyDict['content'] = f.read(20971520)
```

Comment:
what is the reason for this number?

Reply:
It tells python to stop reading the extracted content after approx. 21 MB to avoid huge files in the browser. The number itself was chosen "randomly", open for suggestions :)

Reply:
this is only useful for text files (job logs, etc..) and those are in the worst case ~2-3MB at most ;-) you can drastically limit this so we don't load excess stuff from storage.

Reply:
OK then I'll set the limit to 3MB. The worst case I stumbled upon was a file (athfile-cache.ascii.gz, compressed file in a log archive, size 700KB) that blows up to 22,3MB when the script decompresses it.. still (kind of) a text file that can be displayed though

Reply:
ah, wait, is this number for the compressed or uncompressed size?

Reply:
uncompressed :)

Reply:
ok, then put it at a nice comp-sci number ;-) 32MB or so